gradle-guide-error-prone/src/test/java/com/palantir/gradle/guide/errorprone/IllegalMethodCalledDuringTaskExecutionTest.java

```diff
@@ -84,6 +84,103 @@ final void action() {
             """);
         }
 
+        @Test
+        void fix_whatever_it_can() {
+            test_fix(
+                    "CustomTask.java",
+                    """
+                import org.gradle.api.DefaultTask;
+                import org.gradle.api.tasks.TaskAction;
+                import org.gradle.api.logging.Logger;
+
+                abstract class CustomTask extends DefaultTask {
+                    @TaskAction
+                    final void action() {
+                        getProject();
+                        Logger logger = getProject().getLogger();
+                    }
+                }
+            """,
+                    """
+                import org.gradle.api.DefaultTask;
+                import org.gradle.api.tasks.TaskAction;
+                import org.gradle.api.logging.Logger;
+
+                abstract class CustomTask extends DefaultTask {
+                    @TaskAction
+                    final void action() {
+                        // BUG: Diagnostic contains: Don't call `getProject()` in task actions
```

Comment:
not familiar with the library, but should you not be annotating the before? I suppose this is true for the after as well in this particular case.